app/src/main/java/com/infomaniak/drive/ui/fileList/fileShare/FileShareDetailsFragment.kt

```diff
@@ -62,16 +64,19 @@ class FileShareDetailsFragment : Fragment() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        fileShareViewModel.fetchCurrentFile(navigationArgs.fileId).observe(viewLifecycleOwner) { file ->
-            file?.let { fileShareViewModel.currentFile.value = it } ?: run {
+        fileShareViewModel.fetchCurrentFile(navigationArgs.fileId).observe(viewLifecycleOwner) { currentFile ->
+            currentFile?.let {
+                fileShareViewModel.currentFile.value = it
+                file = it
+            } ?: run {
```

Comment:
Another way to do it

```suggestion
            file = currentFile?.also {
                fileShareViewModel.currentFile.value = it
            } ?: run {
```